wgpu-hal/src/metal/mod.rs

```diff
@@ -929,6 +932,7 @@ struct CommandState {
     index: Option<IndexState>,
     raw_wg_size: MTLSize,
     stage_infos: MultiStageData<PipelineStageInfo>,
+    residency_sets: HashSet<Retained<ProtocolObject<dyn MTLResidencySet>>>,
```

Comment:
@Vecvec we don't need extra interfaces like `set_command_buffer_blas_dependencies` because we can just store that in the builder state here, from which command buffers are then emitted alongside residency sets.